ci2/ci/github.py

```diff
@@ -178,6 +139,45 @@ def from_gh_json(gh_json):
 
         self.state = total_state
 
+    async def heal(self, batch, run, seen_batch_ids):
+        if self.batch is None:
+            batches = await batch.list_batches(
+                complete=False,
+                attributes={
+                    'source_sha': self.source_sha,
+                    'target_sha': self.target_branch.sha
+                })
+
+            # FIXME
+            def batch_was_cancelled(batch):
+                status = batch.status()
+                return any(j['state'] == 'Cancelled' for j in status['jobs'])
+
+            # we might be returning to a commit that was only partially tested
+            batches = [b for b in batches if not batch_was_cancelled(b)]
+
+            # should be at most one batch
+            if len(batches) > 0:
+                self.batch = batches[0]
+            elif run:
+                self.batch = await batch.create_batch(
+                    attributes={
+                        'target_branch': self.target_branch.branch.short_str(),
+                        'pr': str(self.number),
+                        'source_sha': self.source_sha,
+                        'target_sha': self.target_branch.sha
+                    })
+
+                # FIXME build
+                await self.batch.create_job('alpine', ['echo', 'foo'])
```

Comment:
we should be setting the attributes here yeah?

Reply:
I don't need them.  I only look up the batch.  I don't think the CI will ever be concerned with individual jobs (except for logging, etc.)